File: SpEditText/src/main/java/com/sunhapper/x/spedit/view/SpXTextView.java
package com.sunhapper.x.spedit.view;

import android.content.Context;
import android.text.NoCopySpan;
import android.util.AttributeSet;

import com.sunhapper.x.spedit.gif.watcher.GifWatcher;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by sunhapper on 2019/1/25 .
 */
public class SpXTextView extends android.support.v7.widget.AppCompatTextView {

    public SpXTextView(Context context) {
        super(context);
    }

    public SpXTextView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public SpXTextView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    {
        List<NoCopySpan> watchers = new ArrayList<>();
        watchers.add(new GifWatcher(this));
        setSpannableFactory(new SpXSpannableFactory(watchers));
    }

    public void setText(CharSequence text, BufferType type) {
        super.setText(text, BufferType.SPANNABLE);
    }

}

File: SpEditText/src/main/java/com/sunhapper/x/spedit/view/SpXTextView.kt
package com.sunhapper.x.spedit.view

import android.content.Context
import android.text.NoCopySpan
import android.util.AttributeSet
import com.sunhapper.x.spedit.gif.watcher.GifWatcher
import java.util.*

/**
 * Created by sunhapper on 2019/1/25 .
 */
class SpXTextView : android.support.v7.widget.AppCompatTextView {

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    init {
        val watchers = ArrayList<NoCopySpan>()
        watchers.add(GifWatcher(this))
        setSpannableFactory(SpXSpannableFactory(watchers))
    }

    override fun setText(text: CharSequence, type: BufferType) {
        super.setText(text, BufferType.SPANNABLE)
    }

}
